golang/server/server.go

```diff
@@ -114,21 +117,45 @@ func (server MinimalGRPCServer) RunUntilStopped(stopper <-chan struct{}) error {
 	}
 
 	grpcServerResultChan := make(chan error)
+	mux := cmux.New(listener)
+	grpcWebL := mux.Match(cmux.HTTP1Fast())
+	grpcL := mux.Match(cmux.Any())
+
+	go func() {
+		if resultErr := grpcServer.Serve(grpcL); resultErr != nil {
+			logrus.Debugf("error ocurred while creating grpc server: %v", resultErr)
+			grpcServerResultChan <- resultErr
+		}
+	}()
+
+	// grpcweb is a proxy which enables ui to make requests to grpc server using kurtosis-sdk
+	// it converts http/1 request to http/2 and vice-versa under the hood
+	grpcWebServer := grpcweb.WrapServer(grpcServer, grpcweb.WithOriginFunc(func(origin string) bool { return true }))
```

Comment:
Is there a possible race condition here? Does this depend on grpcServer running?

Reply:
i ask cause the gprcServer is serving on a separate go routine

Reply:
Is this Cors policy too open?

Reply:
hmm i think it's fine for v0... because frontend is only being accessible on your local laptop... these optimizations can be done later one. I am focused mainly on the initial e2e functionality - rest can be tackled... will do this in next iteration @h4ck3rk3y . great callout.